Fix feature AUTOINVISIBLE disallowing remote users to set -i, only enforce this on local users.

diff -r 71fe467decbf ircd/s_user.c
--- a/ircd/s_user.c	Sat Jun 29 18:48:55 2013 +0200
+++ b/ircd/s_user.c	Sat Jun 29 18:49:28 2013 +0200
@@ -1350,8 +1350,7 @@
         if (what == MODE_ADD)
           SetInvisible(sptr);
         else
-          if (!feature_bool(FEAT_AUTOINVISIBLE) || IsOper(sptr)) /* Don't allow non-opers to -i if FEAT_AUTOINVISIBLE is set */
-            ClearInvisible(sptr);
+          ClearInvisible(sptr);
         break;
       case 'd':
         if (what == MODE_ADD)
@@ -1469,6 +1468,12 @@
     if (!FlagHas(&setflags, FLAG_PARANOID) && !(IsOper(sptr) && HasPriv(sptr, PRIV_PARANOID)))
       ClearParanoid(sptr);
 
+    /* disallow ordinary users to do MODE -i when feature AUTOINVISIBLE is enabled */
+    if (FlagHas(&setflags, FLAG_INVISIBLE) && !IsInvisible(sptr) &&          /* MODE -i */
+        feature_bool(FEAT_AUTOINVISIBLE) &&                                  /* AUTOINVISIBLE is enabled */
+        !FlagHas(&setflags, FLAG_OPER) && !FlagHas(&setflags, FLAG_LOCOP))   /* (was) not opered */
+       SetInvisible(sptr);
+
     /*
      * only send wallops to opers
      */
